parse/parse.go

```diff
@@ -625,7 +638,35 @@ func isFallback(keys []VariantKey) bool {
 	return true
 }
 
-func (p *parser) parseMatcher() (Matcher, error) {
+func hasAnnotation(variable Variable, declarations []Declaration) bool {
+	for _, v := range declarations {
+		switch t := v.(type) {
+		case InputDeclaration:
+			if t.Operand != variable {
+				continue
+			}
+
+			return t.Annotation != nil
+		case LocalDeclaration:
+			if t.Variable != variable {
+				continue
+			}
+
+			if t.Expression.Annotation != nil {
+				return true
+			}
```

Comment:
We require an annotation with a function. Annotation has optional function.

> Missing Selector Annotation:
> Each selector MUST be a variable that directly or indirectly references a declaration with a function.

Reply:
I looked at the code and annotation is a function. MF2 does not use annotation anymore.

Rename to Function in another PR?!

Ignore me for now!

Reply:
Annotation is the function. The name remains from previous iterations where annotation was also a private or reserved stuff.